src/astro/files/locations/gcs.py

```diff
@@ -0,0 +1,34 @@
+from typing import Dict, List
+from urllib.parse import urlparse, urlunparse
+
+from astro.constants import FileLocation
+from astro.files.locations.base import Location
+from astro.utils.dependencies import GCSClient, GCSHook, gcs
+
+
+class GS(Location):
+    """Handler GS object store operations"""
+
+    location_type = FileLocation.GS
+
+    def get_transport_params(self) -> Dict:
+        """get GCS credentials for storage"""
+        if self.conn_id:
+            gcs_hook = GCSHook(self.conn_id)
+            client = gcs_hook.get_conn()  # type: ignore
+        else:
+            client = GCSClient()
+
+        return {"client": client}
+
+    def get_paths(self) -> List[str]:
```

Comment:
Since we are using "resolve" in the description, do you think `resolve_path` might be better? (I am not sure myself)

Reply:
I think Something like Uniform Resource Identifier(URI) makes more sense because we have a URL as well as a local path here. WDYT?

Proposal - resolve_uri()

Reply:
Naming is always so hard :see_no_evil: 
I'd vote for `list_uris` or `list_filepaths`